controllers/transforms_test.go

```diff
@@ -869,6 +939,62 @@ func TestTransformKataManager(t *testing.T) {
 				},
 			}).WithPullSecret("pull-secret").WithPodAnnotations(map[string]string{"nvidia.com/kata-manager.last-applied-hash": "1929911998"}).WithHostPathVolume("kata-artifacts", "/var/lib/kata", newHostPathType(corev1.HostPathDirectoryOrCreate)).WithHostPathVolume("containerd-config", "/etc/containerd", newHostPathType(corev1.HostPathDirectoryOrCreate)).WithHostPathVolume("containerd-socket", "/run/containerd", nil),
 		},
+		{
+			description: "transform kata manager with custom container runtime socket",
+			ds:          NewDaemonset().WithContainer(corev1.Container{Name: "nvidia-kata-manager"}),
+			cpSpec: &gpuv1.ClusterPolicySpec{
+				KataManager: gpuv1.KataManagerSpec{
+					Repository:       "nvcr.io/nvidia/cloud-native",
+					Image:            "kata-manager",
+					Version:          "v1.0.0",
+					ImagePullPolicy:  "IfNotPresent",
+					ImagePullSecrets: []string{"pull-secret"},
+					Args:             []string{"--test-flag"},
+					Config: &kata_v1alpha1.Config{
+						ArtifactsDir: "/var/lib/kata",
+					},
+					Env: []gpuv1.EnvVar{
+						{
+							Name: "CONTAINERD_CONFIG", Value: "/var/lib/rancher/k3s/agent/etc/containerd/config.toml",
+						},
+						{
+							Name: "CONTAINERD_SOCKET", Value: "/run/k3s/containerd/containerd.sock",
+						},
+						{
+							Name: "CONTAINERD_RUNTIME_CLASS", Value: "nvidia",
+						},
+						{
+							Name: "CONTAINERD_SET_AS_DEFAULT", Value: "true",
+						},
+					},
+				},
+			},
+			expectedDs: NewDaemonset().WithContainer(corev1.Container{
+				Name:            "nvidia-kata-manager",
+				Image:           "nvcr.io/nvidia/cloud-native/kata-manager:v1.0.0",
+				ImagePullPolicy: corev1.PullIfNotPresent,
+				Args:            []string{"--test-flag"},
+				Env: []corev1.EnvVar{
+					{Name: "KATA_ARTIFACTS_DIR", Value: "/var/lib/kata"},
+					{Name: "CONTAINERD_CONFIG", Value: "/runtime/config-dir/config.toml"},
```

Comment:
Question: I notice the order of the envvars are different than in the previous test case. What is the reason for that?

Reply:
The `transformForRuntime` method sets environment variables in addition to the environment variables supplied through the ClusterPolicy. Since this method call was moved, the order of env vars was also changed in the rendered daemonset